### PR TITLE
Chore: upgrade to msw v2

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- upgrades to msw@v2
- migration docs are here: https://mswjs.io/docs/migrations/1.x-to-2.x
- sibling enterprise PR: https://github.com/grafana/grafana-enterprise/pull/6264

**Why do we need this feature?**

- gotta keep updating!
- api is a little nicer 👍 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

based off the renovate PR [here](https://github.com/grafana/grafana/pull/82247), but split it out into it's own PR as there are a lot of changes

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ module.exports = {
   testTimeout: 30000,
   resolver: `<rootDir>/public/test/jest-resolver.js`,
   setupFilesAfterEnv: ['./public/test/setupTests.ts'],
+  testEnvironmentOptions: {
```
Comment (ashharrison90) [inline jest.config.js:37]: this is the recommendation from the author of msw: see https://github.com/mswjs/msw/issues/1786#issuecomment-1782559851

little nervous about this change as it affects all the tests 😬

File: public/app/features/alerting/unified/mockApi.ts [hunk 2]
```diff
@@ -190,85 +190,74 @@ export function mockApi(server: SetupServer) {
...
-                ctx.status(200),
-                ctx.json<NewOnCallIntegrationDTO>({
-                  id: integrationId,
-                  integration: body.integration,
-                  integration_url: `https://oncall-endpoint.example.com/${integrationId}`,
-                  verbal_name: body.verbal_name,
-                  connected_escalations_chains_count: 0,
-                })
-              );
+              return HttpResponse.json({
```
Comment (konrad147) [inline public/app/features/alerting/unified/mockApi.ts:254]: Missing `<NewOnCallIntegrationDTO>` type

File: public/app/features/alerting/unified/components/rules/RuleDetails.test.tsx [hunk 3]
```diff
@@ -42,7 +45,13 @@ const ui = {
   },
 };
 
-const server = setupServer();
+const server = setupServer(
+  http.get(`/api/plugins/${SupportedPlugin.Incident}/settings`, async () => {
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rules/RuleDetails.test.tsx:49]: This endpoint wasn't mocked before migration. Why it's necessary now?

Comment (ashharrison90) [reply public/app/features/alerting/unified/components/rules/RuleDetails.test.tsx:49]: honestly i have no idea 😅 but without it, there's a console error about trying to call an endpoint that hasn't been mocked

**Review by jackw — APPROVED** [review]
Plugin loader part of this PR looks good to me! Awesome work @ashharrison90 ! 🚀

File: public/app/features/alerting/unified/components/rules/RuleDetails.test.tsx (hunk 3 above)
Comment (konrad147) [reply public/app/features/alerting/unified/components/rules/RuleDetails.test.tsx:49]: Ah, probably we missed that earlier 😅  good catch!

**Review by konrad147 — APPROVED** [review]
Alerting part LGTM!

**Pull request merged**